### PR TITLE
Fix: Affix target add listeners protect (#45291)

### LINKED ISSUE #45291: bug: Affix not work when set target
### Reproduction link

[https://ant.design/components/affix-cn#affix-demo-target](https://ant.design/components/affix-cn#affix-demo-target)

### Steps to reproduce

应该是我重构 FC 的时候改出来的问题

https://ant.design/components/affix-cn#affix-demo-target

### What is expected?

Affix 正常工作

### What is actually happening?

Affix 不工作

| Environment | Info |
| --- | --- |
| antd | 5.10.0 |
| React | 18.2.0 |
| System | mac |
| Browser | Chreome |

### PR DESCRIPTION
**Pull request by mingming-ma**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

fix #45291 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Affix bug fix set target problem due to listenerTarget is null   |
| 🇨🇳 Chinese |  修复Affix由于 listenerTarget 为空导致无法set target的问题     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 371e7fc</samp>

Fix affix position bug when scroll container changes. Add guard clause to `InternalAffix` component in `components/affix/index.tsx` to handle undefined or null `listenerTarget` prop.

### <samp>🤖 Generated by Copilot at 371e7fc</samp>

*  Prevent errors when `listenerTarget` prop is undefined or null in `InternalAffix` component ([link](https://github.com/ant-design/ant-design/pull/45314/files?diff=unified&w=0#diff-18949a009c0a2f7f47a0136b5808eade7a157b3d78661edc0f75463594e2ad5dR200))

### CODE REVIEW

File: components/affix/index.tsx [hunk 1]
```diff
@@ -197,6 +197,7 @@ const InternalAffix = React.forwardRef<AffixRef, InternalAffixProps>((props, ref
 
   const addListeners = () => {
     const listenerTarget = targetFunc?.();
+    if (!listenerTarget) return;
```
Comment (afc163) [inline components/affix/index.tsx:200]:
```suggestion
    if (!listenerTarget) {
      return;
    }
```

Comment (afc163) [reply components/affix/index.tsx:200]: 补充一下测试用例？

Comment (li-jia-nan) [reply components/affix/index.tsx:200]: @zombieJ 求支援

**Comment by mingming-ma** [comment]
感谢各位大佬 ，关于测试用例部分还在继续研究中

**Comment by li-jia-nan** [comment]
> 感谢各位大佬 ，关于测试用例部分还在继续研究中

这个场景不太好 mock，我也不知道怎么补，先合了搭个顺风车，用例后面需要 @zombieJ @MadCcc 两位大佬看一下

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**